homeassistant/components/litterrobot/binary_sensor.py

```diff
@@ -0,0 +1,50 @@
+"""Support for Litter-Robot binary sensors."""
+from pylitterbot.enums import LitterBoxStatus
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import LitterRobotEntity
+from .hub import LitterRobotHub
+
+
+class LitterRobotBinarySensor(LitterRobotEntity, BinarySensorEntity):
+    """Litter-Robot binary sensor."""
+
+
+class LitterRobotTimingModeSensor(LitterRobotBinarySensor):
+    """Litter-Robot timing mode sensor."""
+
+    @property
+    def is_on(self) -> bool:
+        """If the binary sensor is currently on or off."""
+        return self.robot.status == LitterBoxStatus.CAT_SENSOR_TIMING
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend, if any."""
+        return "mdi:timer-outline" if self.is_on else "mdi:timer-off-outline"
```

Comment:
IMO, it would be more beneficial to expose the status as a separate sensor which would also be useful for any of the statuses rather than creating one-off binary sensors for each.